### PR TITLE
Typescript Migration: Activity

### PR DESCRIPTION
**Pull request by markxiong0122**
migrated activity component and related page and test files

### CODE REVIEW

**Comment by jcscottiii** [comment]
Take a look at some of these suggestions in this commit https://github.com/jcscottiii/chromium-dashboard/commit/cf0b6bffc0412c4452d9b0b027e8372ac35d4776

**Review by DanielRyanSmith — APPROVED** [review]
Making great progress!

File: client-src/js-src/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+interface FeatureLink {
+  url: string;
+  type: string;
+  information: any;
+  http_error_code: number;
+}
+
+declare interface ChromeStatusClient {
```
Comment (jyasskin) [inline client-src/js-src/types.d.ts:25]: Now that #3991 is in, I think you can replace this file with `import type {ChromeStatusClient} from '../js-src/cs-client.js';` in the places that use that type.

File: client-src/js-src/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  constructor(
+    message: string,
+    resource: string,
+    method: string,
+    status: number
+  );
+}
+
+declare class FeatureNotFoundError extends Error {
+  featureID: number;
```
[the remark this answers is not in the document]
Comment (markxiong0122) [reply client-src/js-src/types.d.ts:14]: good point. I just checked and it should be ID indeed

File: client-src/js-src/types.d.ts (hunk 1 above)
Comment (markxiong0122) [reply client-src/js-src/types.d.ts:25]: made my life easier. Thank you for doing this

**Pull request merged**